diff --git a/anitya/ui.py b/anitya/ui.py
index 4dafa88..8e52a24 100644
--- a/anitya/ui.py
+++ b/anitya/ui.py
@@ -328,7 +328,6 @@ def distro_projects_search(distroname, pattern=None):
 
 
 @APP.route('/project/new', methods=['GET', 'POST'])
-@login_required
 def new_project():
 
     plugins = anitya.lib.plugins.load_plugins(SESSION)
@@ -354,7 +353,7 @@ def new_project():
                 version_url=form.version_url.data.strip() or None,
                 version_prefix=form.version_prefix.data.strip() or None,
                 regex=form.regex.data.strip() or None,
-                user_id=flask.g.auth.openid,
+                user_id=0,
                 check_release=form.check_release.data,
             )
             SESSION.commit()
