diff --git a/files/anitya_cron.py b/files/anitya_cron.py
index be904f9..dd57e1f 100644
--- a/files/anitya_cron.py
+++ b/files/anitya_cron.py
@@ -32,6 +32,9 @@ def indexed_listings():
             # Not all backends have the check_feed classmethod implemented,
             # and that's okay...  Just ignore them for now.
             continue
+        except Exception as e:
+            LOG.error("Caught exception while checking backend %r: %s - %s",
+                      backend, type(e), e)
 
 
 def projects_by_feed(session):
